Release/1.3.0

HeJones opened the pull request:
### What

- Added in feature test library and example feature test for Hello endpoint
- Additional commits related to dependancy updates

### How to review

- Check only commits related to these changes

### Who can review

Anyone

sully90 commented on api/response_writer.go:
```diff
@@ -4,36 +4,52 @@ import (
+
+	createIdentityResponse = JSONResponseWriter{
+		ErrFailedToUnmarshalRequestBody: http.StatusInternalServerError,
+		ErrFailedToReadRequestBody:      http.StatusInternalServerError,
```
Is being unable to read the request body a 500 internal server error or a 400 bad request?

daiLlew replied:
Fair point, will update it to BadRequest

HeJones committed:
Add component test library

jon-acker committed:
Add sample component test

jon-acker committed:
Add example of using predefined Then step

mwnicks committed:
add ci component test files

HeJones committed:
Update dependancies

HeJones committed:
Update go version and dependancies

nshumoogum committed:
Remove second retrieval of environment configurations

Eventhough unlikely this is bad practice as the state of the environment variables
could have changed between the two retrievals, resulting in the app behaving differently
depending on what is being processed.

Instead pass environment variables down the app tree.

nshumoogum committed:
Move initialisation of service in component test suite

Rename files and methods

nshumoogum committed:
Tidy go dependencies

nshumoogum committed:
Increase the time for the service to successfully shutdown in unit test

Currently results in intermittent failure as the time difference set in test is to small
(1ms), upping to 9ms.

HeJones committed:
Update dependancy

HeJones committed:
Update dependancies

nshumoogum committed:
Merge branch 'feature/fix-race-condition' into feature/add-component-tests

HeJones committed:
Tidy dependencies

nshumoogum committed:
Fix up dependencies

HeJones committed:
Remove additional empty lines

HeJones committed:
Merge branch 'feature/add-component-tests' into develop

HeJones committed:
Merge branch 'develop' into release/1.3.0

mike-kelly-methods reviewed (approved):
👍
